### PR TITLE
Asm.js Simd.js : Fix for simd shr sequence cloberring the source reg value

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

**Comment by Krovatkin** [comment]
fixes #940

**Comment by Krovatkin** [comment]
@nmostafa  FYI a PR submitted for #940 (msft) / # 95 (intel-chakra)

**Comment by nmostafa** [comment]
@MikeHolman , can you review this ?

File: lib/Runtime/Language/AsmJsModule.cpp [hunk 1]
```diff
@@ -1661,8 +1661,8 @@ namespace Js
...
-        simdFunctions[AsmJsSIMDBuiltin_int8x16_shiftLeftByScalar]   = SIMDFunc(PropertyIds::shiftLeftByScalar, Anew(&mAllocator, AsmJsSIMDFunction, nullptr, &mAllocator, 2, AsmJsSIMDBuiltin_int8x16_shiftLeftByScalar, OpCodeAsmJs::Simd128_ShLtByScalar_I16, AsmJsRetType::Int8x16, AsmJsType::Int8x16, AsmJsType::Unsigned));
-        simdFunctions[AsmJsSIMDBuiltin_int8x16_shiftRightByScalar]  = SIMDFunc(PropertyIds::shiftRightByScalar, Anew(&mAllocator, AsmJsSIMDFunction, nullptr, &mAllocator, 2, AsmJsSIMDBuiltin_int8x16_shiftRightByScalar, OpCodeAsmJs::Simd128_ShRtByScalar_I16, AsmJsRetType::Int8x16, AsmJsType::Int8x16, AsmJsType::Unsigned));
+        simdFunctions[AsmJsSIMDBuiltin_int8x16_shiftLeftByScalar]   = SIMDFunc(PropertyIds::shiftLeftByScalar, Anew(&mAllocator, AsmJsSIMDFunction, nullptr, &mAllocator, 2, AsmJsSIMDBuiltin_int8x16_shiftLeftByScalar, OpCodeAsmJs::Simd128_ShLtByScalar_I16, AsmJsRetType::Int8x16, AsmJsType::Int8x16, AsmJsType::Int));
```
Comment (MikeHolman) [inline lib/Runtime/Language/AsmJsModule.cpp]: Does this conflict with #934?

Comment (nmostafa) [inline lib/Runtime/Language/AsmJsModule.cpp]: I don't see how. We are relaxing the shift amount type from Unsigned to Int. #934 allows coercion of SIMD return types.

Comment (MikeHolman) [inline lib/Runtime/Language/AsmJsModule.cpp]: Ah you're right, was misreading that as the ret type!

**Comment by MikeHolman** [comment]
lgtm

**Pull request merged**